packages/superset-ui-query/src/buildQueryObject.ts

```diff
@@ -42,7 +43,7 @@ export default function buildQueryObject<T extends QueryFormData>(formData: T):
     metrics: processMetrics(formData),
     order_desc: typeof order_desc === 'undefined' ? true : order_desc,
     orderby: [],
-    row_limit: Number(row_limit),
+    row_limit: row_limit == null || isNaN(numericRowLimit) ? undefined : numericRowLimit,
```

Comment:
Should use `Number.isNaN`, instead of `isNaN` 

Reply:
I'm fixing this in my test coverage PR: https://github.com/apache-superset/superset-ui/pull/390/files

Reply:
Thank you!